cli/utilities/containers.py

```diff
@@ -86,7 +88,7 @@ def run(
             user (str): User to run container as
             entry_point (str): Entry point command or executable to be run
         """
-        if not image and not rm:
+        if not image and not (rm or rmi):
```

Comment:
You may also want to add another check for `rm and rmi`... if `true` raise an exception.

Reply:
Sorry to provide a bit more context. I notice that you have the list of images passed via `rmi` however podman supports the sub command `rm | rmi` both are for image removal.

Having `rmi` as an option may cause confusion later unless we review ln:133 which effectively uses only `rm`

If I understand correctly, you want support for multiple images... if that is the case, the use `image` variable and pass multiple image names separated by `space`

Reply:
Hey @psathyan 

I added the changes to accommodate running the below 2 podman run commands without causing a failure -
First command is run as part of https://github.com/red-hat-storage/cephci/blob/master/cephci/utils/configure.py#L330
```
#podman run -v /opt/registry/certs:/certs:Z -v /opt/registry/certs/domain.cert:/certs/domain.cert:Z --rm registry.redhat.io/rhel8/skopeo:8.5-8 skopeo copy --remove-signatures --src-creds qa@redhat.com:MTQj5t3n5K8
6p3gH --dest-cert-dir=./certs/ --dest-creds myregistryusername:myregistrypassword docker://registry.redhat.io/openshift4/ose-prometheus-node-exporter:latest docker://ceph-saya-auto-qepuhn-node1-installer:5000/openshift4/ose-prometheus-node-exporter:latest
```

and the second command is 
`# podman run -i --rm --user root --entrypoint /bin/bash registry.redhat.io/openshift4/ose-prometheus:v4.15 -c "prometheus --version"`
(https://github.com/red-hat-storage/cephci/blob/master/tests/cephadm/test_ceph_orch_for_monitoring_image_version.py#L25)

Where in first command we pass a image to --rm flag and in second we don't, instead we want the container to be removed after the use.
Please let me know what modifications need to be added.
Thank you.

Reply:
Thanks @sayaleeraut for the detail explanation.

```
# podman run \
    -v /opt/registry/certs:/certs:Z \
    -v /opt/registry/certs/domain.cert:/certs/domain.cert:Z \
    --rm \
    registry.redhat.io/rhel8/skopeo:8.5-8 \
    skopeo \.  # <- this is arguments to image
        copy \
        --remove-signatures \
        --src-creds xxx \
        --dest-cert-dir=./certs/ \
        --dest-creds myregistryusername:myregistrypassword \
        docker://registry.redhat.io/openshift4/ose-prometheus-node-exporter:latest \
        docker://ceph-saya-auto-qepuhn-node1-installer:5000/openshift4/ose-prometheus-node-exporter:latest
```

IMO, we shouldn't be using `start_local_private_registry` method instead have a separate one as this is specific to a copy.

Having a separate function that simulates an image transfer / copy would help. wdyt?

Reply:
Hey @psathyan 
We are using `add_images_to_private_registry` for this transfer/copy purpose (https://github.com/red-hat-storage/cephci/blob/master/cephci/utils/configure.py#L29)

And that one is calling Container(node).run(); which I think is the correct code flow.
Thanks!

Reply:
My bad, I should checked correctly. I missed line: 293 and went to `start_local_private_registry` definition on ln: 264. This side tracked the discussion. Sorry about it.

Your earlier comments had the details that I should have read correctly. 

As discussed, we can remove `rmi` and set `image` in add method. Thanks for debugging this with me.

Reply:
Hey @psathyan Thanks for reviewing.
I have added the discussed changes, please review.

Reply:
Thanks @sayaleeraut